src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyout.cs

```diff
@@ -80,31 +82,314 @@ public IList<MenuFlyoutItemBase> Items
+			}
+			else
+			{
+				m_openWindowed = true;
+			}
+
+			var placementTarget = pPlacementTarget;
+			CacheInputDeviceTypeUsedToOpen(placementTarget);
+		}
+
+		// Raise Opening event.
+		private protected override void OnOpening()
+		{
+			// Update the TemplateSettings as it is about to open.
+			Control presenter = GetPresenter();
+			MenuFlyoutPresenter menuFlyoutPresenter = presenter as MenuFlyoutPresenter;
+
+			if (menuFlyoutPresenter != null)
+			{
+				IMenu parentMenu = (this as IMenu).ParentMenu as IMenu;
+
+				(menuFlyoutPresenter as IMenuPresenter).OwningMenu = parentMenu != null ? parentMenu : this;
+				menuFlyoutPresenter.UpdateTemplateSettings();
+			}
+
+			// Reset the presenter's ItemsSource.  Since Items is not an IObservableVector, we don't
+			// automatically respond to changes within the vector.  Clearing the property when the presenter
+			// unloads and resetting it before we reopen ensures any changes to Items are reflected
+			// when the MenuFlyoutPresenter shows.  It also allows sharing of MenuFlyouts; since MenuFlyoutItemBases
+			// are UIElements they must be unparented when leaving the tree before they can be inserted elsewhere.
+			menuFlyoutPresenter.ItemsSource = m_tpItems;
+
+			AutomationPeer.RaiseEventIfListener(menuFlyoutPresenter, AutomationEvents.MenuOpened);
+		}
+
+		private protected override void OnClosing(ref bool cancel)
+		{
+			base.OnClosing(ref cancel);
+
+			if (!cancel)
+			{
+				CloseSubMenu();
+			}
+		}
+
+		private protected override void OnClosed()
+		{
+			CloseSubMenu();
+
+			AutomationPeer.RaiseEventIfListener(GetPresenter(), AutomationEvents.MenuClosed);
+
+			((MenuFlyoutPresenter)GetPresenter()).m_iFocusedIndex = -1;
+			((ItemsControl)GetPresenter()).ItemsSource = null;
+		}
+
+		void CloseSubMenu()
+		{
+			MenuFlyoutPresenter presenter = (MenuFlyoutPresenter)GetPresenter();
+
+			if (presenter != null)
+			{
+				IMenuPresenter subPresenter = (presenter as IMenuPresenter).SubPresenter;
+
+				if (subPresenter != null)
+				{
+					subPresenter.CloseSubMenu();
+				}
+			}
+		}
+
+		void PreparePopupTheme(
+			Popup pPopup,
+			MajorPlacementMode placementMode,
+			FrameworkElement pPlacementTarget)
+		{
+			bool areOpenCloseAnimationsEnabled = false;
+			areOpenCloseAnimationsEnabled = AreOpenCloseAnimationsEnabled;
+
+			if (!areOpenCloseAnimationsEnabled)
+			{
+				return;
+			}
+
+			// On Threshold, we use the MenuPopupThemeTransition for MenuFlyouts.
+			double openedLength = 0;
+
+			if (m_tpMenuPopupThemeTransition == null)
+			{
+				// UNO TODO
+				//Transition spMenuPopupChildTransition;
+				//spMenuPopupChildTransition = PreparePopupThemeTransitionsAndShadows(pPopup, 0.5 /* closedRatioConstant */, 0 /* depth */);
+				//m_tpMenuPopupThemeTransition = spMenuPopupChildTransition;
+			}
+
+			openedLength = ((Control)GetPresenter()).ActualHeight;
+
+			// UNO TODO
+			// (m_tpMenuPopupThemeTransition as MenuPopupThemeTransition).OpenedLength = openedLength;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)